### ISSUE TITLE
Page gets wiped w/ Scroll To Top Button enabled in Firefox

### ISSUE THREAD

**Issue by PoziWorld** [LessWrong2/Lesswrong2#2657]
Hello there, 

I am a developer of the [Scroll To Top Button browser extension](https://github.com/PoziWorld/Scroll-To-Top-Button-extension). 

Recently, @markran reported an [issue](https://github.com/PoziWorld/Scroll-To-Top-Button-extension/issues/6) that happens on LessWrong 2.0 while accessing it in Firefox with the Scroll To Top Button extension enabled:
* When hovering over some elements, “`Error: Error: Permission denied to access property "offsetParent"`” gets displayed in place of some elements.
* When hovering over the left navigation menu, the whole page gets wiped (`<div id="react-app"></div>` loses all of its children elements).

See a [video of the issue](https://streamable.com/92wix).

The issue doesn't seem to happen in Chrome.

<hr>

Scroll To Top Button injects its markup (the buttons and their container) in an unexpected place of the page markup: after `</body>`. 

Per standard, no page elements should be placed outside the body, but some browser extensions employ this technique to make sure their elements, such as the Scroll To Top Button's buttons, are always on top of other elements and to make sure those elements don't get (un)intentionally removed by the page's scripts (for example, there was an issue on Bing: its JavaScript was removing one of the buttons).

If Scroll To Top Button gets injected within the body, the issue doesn't occur.

A rule can be added in Scroll To Top Button that will inject the buttons and their container within the body just on this website and will continue to do otherwise on all other websites, but potentially there are other extensions out there that cause the same issue on this website, so it might be best to fix it on the website.

Would you agree?

**Comment by jimrandomh** [comment]
I tried it out and the issue didn't replicate. Firefox 70.0.1, ie not on the beta channel which the video was recorded with. Scroll to Top v4.8.0.2, released after you made this report. It looks like a workaround has already been applied, so I'd have to go to an older version to reproduce. Anyways, I got this stack trace out of Sentry:

```
meteor://app/packages/modules.js in getOffsetParent at line 180031:5
meteor://app/packages/modules.js in findCommonOffsetParent at line 180105:12
meteor://app/packages/modules.js in getReferenceOffsets at line 180582:83
meteor://app/packages/modules.js in applyStyleOnLoad at line 181031:26
```

These function names are from `popper.js` (incorporated indirectly via `@material-ui`). Looking at the source of `popper.js`, `getOffsetParent` walks a subset of the DOM, including your extension-added element, and assumes that for any DOM element `offsetParent` is either a DOM node, or null. But in Firefox 71, accessing your element's `offsetParent` throws an exception instead. Given that https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent makes no mention of the possibility of throwing exceptions, and that this has so far only been observed in Firefox's beta channel, I believe this is actually a bug in Firefox, and should be reported to them.
